cmd/crank/render/render_test.go

```diff
@@ -349,6 +361,20 @@ func TestRender(t *testing.T) {
 							},
 						},
 					},
+					Conditions: []unstructured.Unstructured{
+						{
+							Object: map[string]any{
+								"apiVersion":         "render.crossplane.io/v1beta1",
+								"kind":               "Condition",
```

Comment:
Are these conditions associated with the XR? Should we return them as part of its status, rather than distinct resources?

Reply:
Good question @negz, I was going back and forth on exposing these as a high level item or attaching them to the XR. I decided on exposing them as a raw high level list because:

* that's how they are returned from [`FunctionComposer.Compose()`](https://github.com/crossplane/crossplane/blob/main/internal/controller/apiextensions/composite/composition_functions.go#L551) and the `render` command experience has mostly seemed to follow that boundary so far.
* if they were to be directly set on the XR as regular ol' conditions then we'd lose the clear indication about if they are targeting the claim too or not.

That said, we could still attach them to the XR and just add a "target" field there too (it's all unstructured 🤪), that could be reasonable. Do you have a strong preference?

Reply:
I expect the output to `crossplane render` to reflect the actual resources that would be composed in a real deployment as closely as possible. In a real deployment conditions would be status conditions on the XR, and optionally the claim. Not distinct resources.

Today `crossplane render` uses "virtual" resources for results and context. I don't love this, but can't think of a better option. Virtual resources are misleading. A new user could easily think there's a real context or result resource - it's not obvious which resources are real and which are virtual.

So I recommend avoiding virtual resources unless there's no other option.

In this case I'd recommend:

* Set regular conditions on the XR
* If showing claim conditions is important include the claim (with its conditions) in the ouput

Reply:
Cool @negz, I think that's a reasonable approach to take here so it's more clear on the end state of the resources. I'll make the updates to set the conditions on the XR.  We don't really have a concept right now of claim handling at all in `render`, so I'll leave that out of scope. Seeing the condition properties/values on the XR will already be a big improvement for function authors working with conditions.

Reply:
@negz @dalton-hill-0 I updated this PR with the approach of just setting the conditions on the XR instead of exposing them as virtual resources. I have updated the PR description with more details about the approach here and the testing results, so check that out too and please let me know what you think! 🙇‍♂️

Reply:
LGTM - Thanks for adding this. This will help developers understand how to use the status-transformer function.

Reply:
@negz let me know if you have any lingering feedback you'd like to see incorporated, thank you!